rom/laghos_rom.cpp

```diff
@@ -3262,6 +3266,197 @@ void ROM_Operator::EvalSpaceTimeResidual_RK4(Vector const& S, Vector &f) const
+        o << "Error indicator: " << greedyErrorIndicatorType << std::endl;
+        o.close();
+    }
+
+    if (parameterPointGreedySampler->isComplete())
+    {
+        // The greedy algorithm procedure has ended
+        MFEM_ABORT("The greedy algorithm procedure has completed!");
+    }
+
+    // First check if we need to compute another error indicator
+    struct CAROM::GreedyErrorIndicatorPoint pointRequiringErrorIndicator = parameterPointGreedySampler->getNextPointRequiringErrorIndicator();
+    CAROM::Vector* errorIndicatorPointData = pointRequiringErrorIndicator.point.get();
+    struct CAROM::GreedyErrorIndicatorPoint pointRequiringRelativeError = parameterPointGreedySampler->getNextPointRequiringRelativeError();
+    CAROM::Vector* samplePointData = pointRequiringRelativeError.point.get();
+    double* greedyParam = getGreedyParam(romOptions, greedyParamString);
+
+    if (errorIndicatorPointData != NULL)
+    {
+        CAROM::Vector* localROM = pointRequiringErrorIndicator.localROM.get();
+        romOptions.basisIdentifier = "_" + to_string(localROM->item(0));
+        *greedyParam = errorIndicatorPointData->item(0);
+
+        double errorIndicatorEnergyFraction = 0.9999;
+
+        char tmp[100];
+        sprintf(tmp, ".%06d", myid);
+
+        std::string fullname = outputPath + "/" + std::string("errorIndicatorVec") + tmp;
+
+        if (romOptions.greedyErrorIndicatorType == varyTimeStep ||
+                romOptions.greedyErrorIndicatorType == varyBasisSize)
+        {
+            t_final = 0.001;
+        }
+
+        std::ifstream checkfile(fullname);
+        if (!checkfile.good())
+        {
+            if (romOptions.greedyErrorIndicatorType == varyTimeStep)
+            {
+                dt_factor = 1.0;
+            }
+            else if (romOptions.greedyErrorIndicatorType == varyBasisSize)
+            {
+                errorIndicatorEnergyFraction = 0.99;
+            }
+        }
+
+        // Get the rdim for the basis used.
+        readNum(romOptions.dimX, outputPath + "/" + "rdimx" + romOptions.basisIdentifier + "_" + to_string(errorIndicatorEnergyFraction));
+        readNum(romOptions.dimV, outputPath + "/" + "rdimv" + romOptions.basisIdentifier + "_" + to_string(errorIndicatorEnergyFraction));
+        readNum(romOptions.dimE, outputPath + "/" + "rdime" + romOptions.basisIdentifier + "_" + to_string(errorIndicatorEnergyFraction));
+        if (!romOptions.SNS)
+        {
+            readNum(romOptions.dimFv, outputPath + "/" + "rdimfv" + romOptions.basisIdentifier + "_" + to_string(errorIndicatorEnergyFraction));
+            readNum(romOptions.dimFe, outputPath + "/" + "rdimfe" + romOptions.basisIdentifier + "_" + to_string(errorIndicatorEnergyFraction));
+        }
+
+        rom_online = true;
+    }
+    else if (samplePointData != NULL)
+    {
+        CAROM::Vector* localROM = pointRequiringRelativeError.localROM.get();
+        romOptions.basisIdentifier = "_" + to_string(localROM->item(0));
```

Comment:
Although it seems unlikely, could two basis identifiers be the same and cause incorrect behavior?

Reply:
Yes. Two identical basis identifiers would be a bug and should not happen. The greedy algorithm has checks to prevent the same point being sampled twice.

Reply:
My question is actually for multidimensional parameters, which are not implemented (yet) in Laghos. The issue is that it only uses `item(0)` for the identifier. For 1D parameters like `bef` that is fine, but in general I suppose it should depend on all entries, to avoid different `localROM` values having the same `basisIdentifier`.